### DISCUSSION TITLE
How to Convert between Character Set Format

### DISCUSSION THREAD

**Discussion by RyanBram** [ImageMagick/ImageMagick#5959] [Help] [answered]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

Windows

### Description

I have a DLC character that I will use for the game engine that I use. However, I found that the character set placement format is different, so I have to make adjustments before it can be used. Because there are so many character sets, it would be a long task to convert them all, so I tried to find a solution with Image Magick.

Please help me how to do the conversion process by using the windows bat file to convert the following Format 1 to Format 2:
| Format 1 | to | Format 2 |
|---|---|---|
| ![Format 1](https://user-images.githubusercontent.com/4508389/211704520-9cfdda5f-c45c-43bc-b3d0-9d9007053911.png) | => | ![Format 2](https://user-images.githubusercontent.com/4508389/211704539-7b85771d-ad40-45ea-9b88-5e51aaa488cc.png) |

**Comment by fmw42** [comment]
There is no attached .bat file.

Crop off the top row. Then crop the rest into 4 parts (top-to-bottom). Then swap them and append them vertically as appropriate.  Sorry, I do not use Windows, so cannot help with a .bat file.

**Reply by RyanBram** [reply]
Pardon my English.

What I mean is, I need an example bat script or windows command line that can convert the above 2 formats.

**Reply by RyanBram** [reply]
Thank you very much for the advice.
Theoretically, I think I understand what you mean. But I'm still confused about the ImageMagick flags that I should use for the above tasks.

If you'd like, an example in the form of a Linux bash command would be fine.

Sorry for the trouble. 🙏

**Comment by fmw42** [comment, marked as answer]
I am not a Windows user, but try this in a .bat script.  I crop into 5 parts top to bottom. Then I copy the 4 parts that you want and re-arrange the copies. Then I delete the original 5 crop parts. Then I append the 4 rearranged parts.

```
magick DLC.png ^
-crop 100x%%[fx:100/5]%% +repage ^
( -clone 3 -clone 4 -clone 2 -clone 1 ) ^
-delete 0-4 ^
-append ^
DLC_new.png

```

The following equivalent in Unix syntax works for me on my Mac.  

```
magick DLC.png \
-crop "100x%[fx:100/5]%" +repage \
\( -clone 3 -clone 4 -clone 2 -clone 1 \) \
-delete 0-4 \
-append \
DLC_new.png

```

**Reply by RyanBram** [reply]
Wow,  it really works 💯 as intended.  🎉✨
My problem really solved.

Thank you very much for your big help. 😃
